src/transformers/models/layoutlmv2/modeling_layoutlmv2.py

```diff
@@ -1424,3 +1462,249 @@ def forward(
+        return loss, all_pred_relations
+
+
+@add_start_docstrings(
+    """
+    LayoutLMv2 Model with a relation extraction head on top for key-value extraction tasks such as
+    [XFUND](https://github.com/doc-analysis/XFUND) (a bi-affine attention layer on top).
+    """,
+    LAYOUTLMV2_START_DOCSTRING,
+)
+class LayoutLMv2ForRelationExtraction(LayoutLMv2PreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.layoutlmv2 = LayoutLMv2Model(config)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+        self.extractor = LayoutLMv2RelationExtractionDecoder(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(LAYOUTLMV2_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
+    @replace_return_docstrings(output_type=RelationExtractionOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids: Optional[torch.LongTensor] = None,
+        bbox: Optional[torch.LongTensor] = None,
+        image: Optional[torch.FloatTensor] = None,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        token_type_ids: Optional[torch.LongTensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        entities: Optional[dict] = None,
+        relations: Optional[dict] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        r"""
+        entities (`list[dict]`):
+            List of dictionaries (one per example in the batch). Each dictionary contains 3 keys: `start`, `end` and
+            `label`.
+        relations (`list[dict]`):
+            List of dictionaries (one per example in the batch). Each dictionary contains 4 keys: `start_index`,
+            `end_index`, `head` and `tail`.
+
+        Returns:
+
+        Example:
+
+        ```python
+        >>> from transformers import LayoutLMv2Processor, LayoutLMv2ForRelationExtraction
+        >>> from PIL import Image
+        >>> from datasets import load_dataset
+
+        >>> processor = LayoutLMv2Processor.from_pretrained("microsoft/layoutlmv2-base-uncased")
+        >>> model = LayoutLMv2ForRelationExtraction.from_pretrained("microsoft/layoutlmv2-base-uncased")
+
+        >>> dataset = load_dataset("hf-internal-testing/fixtures_docvqa")
+        >>> image_path = dataset["test"][0]["file"]
+        >>> image = Image.open(image_path).convert("RGB")
+        >>> encoding = processor(image, return_tensors="pt")
+
+        >>> # instantiate relations as empty at inference time
+        >>> encoding["entities"] = [{"start": [0, 4], "end": [3, 6], "label": [2, 1]}]
+        >>> encoding["relations"] = [{"start_index": [], "end_index": [], "head": [], "tail": []}]
```

Comment:
Model cant take lists as inputs as they then won't work with ONNX/distributed etc. This should all be tensors.